python/ray/tune/function_runner.py

```diff
@@ -457,6 +482,38 @@ def cleanup(self):
             logger.debug("Clearing temporary checkpoint: %s",
                          self.temp_checkpoint_dir)
 
+    def reset_config(self, new_config):
+        if self._runner and self._runner.is_alive():
+            self._end_event.set()
+            self._continue_semaphore.release()
+            # Wait for thread termination so it is save to re-use the same
+            # actor.
+            self._runner.join(timeout=1)
+            if self._runner.is_alive():
+                # Did not finish within timeout, reset unsuccessful.
```

Comment:
maybe use a longer timeout?

Reply:
actually probably best to provide an env var?

Reply:
I added an env var. 2 Seconds should be enough, as the thread should just jump directly into `sys.exit()`